_integration/testsuite/httpproxy/004-https-sni-enforcement.yaml

```diff
@@ -231,42 +214,25 @@ spec:
 
 ---
 
-fqdn := "echo-two.projectcontour.io"
-
-default proxy_is_present = false
-proxy_is_present {
-  data.resources.httpproxies["echo-two"]
-}
+import data.contour.resources
 
 fatal_proxy_is_not_present[msg] {
-  not proxy_is_present
-  msg := sprintf("HTTPProxy for %q is not present", [ fqdn ])
+  not resources.is_present("httpproxies", "echo-two")
+  msg := sprintf("HTTPProxy %q is not present", [ "echo-two" ])
 }
 
 ---
 
-Name := "echo-two"
-
-default proxy_is_valid = false
-proxy_is_valid {
-  proxy := data.resources.httpproxies[Name]
-  proxy.status.currentStatus == "valid"
-}
-
-fatal_proxy_is_not_valid[msg] {
-  not proxy_is_valid
-  msg := "Invalid proxy"
-}
+import data.contour.resources
 
 fatal_proxy_is_not_valid[msg] {
-  proxy := data.resources.httpproxies[Name]
-  fqdn := proxy.spec.virtualhost.fqdn
-  status := proxy.status
+  status := resources.status("httpproxies", "echo-two")
+  validity := object.get(status, "currentStatus", "not-present")
 
-  not proxy_is_valid
+  not validity == "valid"
```

Comment:
So this expects the validity to "not" be `valid`?

Reply:
yep, it's an error in that case.

Reply:
Maybe change `validity` to `resultStatus` or something such that it reads: `not resultStatus == "valid"` or check on the reverse such that it's `invalid`.

Reply:
I renamed it to `resultStatus`.